backend/src/apiserver/server/pipeline_upload_server.go

```diff
@@ -191,6 +217,25 @@ func (s *PipelineUploadServer) UploadPipelineVersion(w http.ResponseWriter, r *h
 	}
 }
 
+func (s *PipelineUploadServer) canUploadVersionedPipeline(r *http.Request, namespace string) error {
+	if namespace == "" {
+		return nil
+	}
+	userIdentityHeader := r.Header.Get(common.GetKubeflowUserIDHeader())
+	resourceAttributes := &authorizationv1.ResourceAttributes{
+		Namespace: namespace,
+		Verb:      common.RbacResourceVerbList,
+		Group:     common.RbacPipelinesGroup,
+		Version:   common.RbacPipelinesVersion,
+		Resource:  common.RbacResourceTypePipelines,
+	}
+	err := s.resourceManager.IsRequestAuthorized(userIdentityHeader, resourceAttributes)
```

Comment:
Why not call `isAuthorized` here, same as in other handlers?

Reply:
```isAuthorized``` takes the Context, which is not available here.